api/src/main/java/com/velocitypowered/api/bossbar/BossBarManager.java

```diff
@@ -0,0 +1,95 @@
+package com.velocitypowered.api.bossbar;
+
+import java.util.Collection;
+import java.util.UUID;
+import net.kyori.text.Component;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Represents a boss bar manager
+ */
+public interface BossBarManager {
+
+  /**
+   * Creates a new {@link BossBar}
+   *
+   * @param title boss bar title
+   * @param color boss bar color
+   * @param overlay boss bar overlay
+   * @return a fresh boss bar
+   */
+  @NonNull
+  BossBar create(@NonNull Component title, @NonNull BossBarColor color,
+      @NonNull BossBarOverlay overlay);
+
+  /**
+   * Creates a new {@link BossBar}
+   *
+   * @param title boss bar title
+   * @param color boss bar color
+   * @param overlay boss bar overlay
+   * @param progress boss bar progress
+   * @return a fresh boss bar
+   * @throws IllegalArgumentException if progress not between 0 and 1
+   */
+  @NonNull
+  BossBar create(
+      @NonNull Component title, @NonNull BossBarColor color, @NonNull BossBarOverlay overlay,
+      float progress);
+
+  /**
+   * Creates a new {@link BossBar}
+   *
+   * @param title boss bar title
+   * @param color boss bar color
+   * @param overlay boss bar overlay
+   * @param progress boss bar progress
+   * @param uuid uuid of the boss bar
+   * @return a fresh boss bar, <b>or null if there's already a boss bar with this uuid</b>
+   * @throws IllegalArgumentException if progress not between 0 and 1
+   */
+  @Nullable
+  BossBar create(
```

Comment:
Why is this `@Nullable`? Why are you allowing users to pass in arbitrary UUIDs? If it's because you're throwing an exception, then throwing an exception implicitly has no return value.